src/indexable.cr

```diff
@@ -809,6 +809,38 @@ module Indexable(T)
     index(offset) { |e| yield e } || raise Enumerable::NotFoundError.new
   end
 
+  # Returns the first element in the indexable for which the passed block
+  # is truthy, starting from the given *offset*.
+  #
+  # Accepts an optional parameter *if_none*, to set what gets returned if
+  # no element is found (defaults to `nil`).
+  #
+  # ```
+  # [1, 2, 3, 4].find { |i| i > 2 }        # => 3
+  # [1, 2, 3, 4].find(-1) { |i| i > 8 }    # => -1
+  # [1, 2, 3, 4].find(-1, 2) { |i| i < 2 } # => -1
+  # ```
+  def find(if_none = nil, offset : Int = 0, & : T ->)
+    offset += size if offset < 0
+    return nil if offset < 0
+
+    if idx = index(offset) { |i| yield i }
+      return unsafe_fetch(idx)
+    end
+    if_none
+  end
+
+  # Returns the first element in the indexable for which the passed block is truthy.
+  # Raises `Enumerable::NotFoundError` if there is no element for which the block is truthy.
+  #
+  # ```
+  # [1, 2, 3, 4].find! { |i| i > 2 } # => 3
+  # [1, 2, 3, 4].find! { |i| i > 8 } # => raises Enumerable::NotFoundError
+  # ```
```

Comment:
Sorry one last thing. We should probably mention the `offset` parameter like we did for `#find`. As well as have some examples showing its usage.

Reply:
great point, addressed